docs/source/opens-upgrade.rst

```diff
@@ -0,0 +1,133 @@
+Upgrade Procedure (OpenStack)
+=============================
+
+This document details the procedure for upgrading a Calico-based OpenStack
+system. It contains the full steps for upgrading all components and the order
+in which that upgrade should be performed. Most releases do not concurrently
+upgrade all of these components: if a release does not upgrade a given
+component, you may skip those steps.
+
+.. warning:: While the upgrade procedure is very safe, you will be unable to
+             issue API requests to your OpenStack system during the procedure.
+             Please plan your upgrade window accordingly, and see the
+             :ref:`service_impact` section for more details.
+
+
+.. _service_impact:
+
+Service Impact
+--------------
+
+During the upgrade, **all VMs will continue to function normally**: there
+should be no impact on the data plane. However, control plane traffic may fail
+at different points throughout the upgrade.
+
+Generally, users should be prevented from creating or updating virtual machines
+during this procedure, as these actions will fail. VM deletion *may* succeed,
+but will likely be delayed until the end of the upgrade.
+
+For this reason, we highly recommend planning a maintenance window for the
+upgrade. During this window, you should disable all user API access to your
+OpenStack deployment.
+
+Upgrade Procedure
+-----------------
+
+Upgrade is performed in the following stages.
```

Comment:
"stages, which must be performed in the order shown."  [Worth hammering home this point, I think.]